### DISCUSSION TITLE
AirSim, ArduPilot and WSL not connecting

### DISCUSSION THREAD

**Discussion by tuvyz** [microsoft/AirSim#4845] [Support Q&A]
Hello! I faced such problem. I built AirSim on Windows, installed WSL and installed ArduPilot on it. However, when I start AirSim it just freezes and won't exit even if ArduPilot is started. Here is the settings.json file:
{
     "SeeDocsAt": "https://github.com/Microsoft/AirSim/blob/main/docs/settings.md",
     "SettingsVersion": 1.2,
     "SimMode": "Multirotor",
     "Origin Geopoint": {
       "latitude": -35.363261,
       "Longitude": 149.165230,
       Altitude: 583
     },
     "Vehicles": {
         "Copter": {
             "VehicleType": "ArduCopter",
             "UseSerial": false,
             "LocalHostIp": "172.18.128.1",
             "UdpIp": "172.18.132.194",
             "UdpPort": 9003,
             "Control Port": 9002
         }
     }
}

And with this command I launch ArduPilot:
sim_vehicle.py -v ArduCopter -f airsim-copter -A "--sim-address=172.18.128.1 --sim-port-in=9003 --sim-port-out=9002"

At the specified addresses, Windows and WSL can ping each other. I turned off Windows Firewall completely. However, this does not work.
At the same time, before that I used PX4 and it worked perfectly in such a bundle. Tell me what could be the problem please

**Comment by ghost** [comment]
{
  "SeeDocsAt": "https://github.com/Microsoft/AirSim/blob/main/docs/settings.md",
  "SettingsVersion": 1.2,
  "SimMode": "Multirotor",
  "OriginGeopoint": {
    "Latitude": 37.334947,
    "Longitude": -122.012715,
    "Altitude": 583
  },
  "Vehicles": {
    "Copter": {
      "VehicleType": "ArduCopter",
      "UseSerial": false,
      "LocalHostIp": "127.0.0.1",
      "UdpIp": "127.0.0.1",
      "UdpPort": 9003,
      "ControlPort": 9002,
      "AutoCreate": true,
      "RC": {
        "RemoteControlID": 2
      }
   }
 }
}

use this in settings.json and run this command on your wsl, it works fine

sim_vehicle.py -v ArduCopter -f airsim-copter --console --map
